### PR TITLE
sequence & callhome: Add sequence automation stats

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- CHANGELOGs > Added notes.
- ExtensionCallHome > Add stats key for telemetry.
- SequenceActiveScanJob > Add stats counters.
- StdActiveScanRunner > Add stats counters.
- SequenceImportJob > Add stats counters.

## Related Issues
N/A

## Checklist
- [na] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [na] Write tests
- [na] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by psiinon** [comment]
This now has conflicts

**Comment by kingthorin** [comment]
Deconflicted. @thc202 I re-requested your review as it meant relocating one of the method calls.

**Review by thc202 — APPROVED** [review]

File: addOns/sequence/src/main/java/org/zaproxy/zap/extension/sequence/automation/SequenceActiveScanJob.java [hunk 1]
```diff
@@ -255,19 +256,33 @@ private static List<SequenceStepData> scanSequence(
                 new StdActiveScanRunner(
                         script, contextWrapper.getContext(), user, contextSpecificObjects);
 
+        Stats.incCounter(ExtensionSequenceAutomation.STATS_PREFIX + "ascan.scan");
```
Comment (psiinon) [inline addOns/sequence/src/main/java/org/zaproxy/zap/extension/sequence/automation/SequenceActiveScanJob.java:259]: Is it possible to move (some of) these stats into the StdActiveScanRunner?
Right now we're not getting any stats if the scan is run from the desktop..

Comment (kingthorin) [reply addOns/sequence/src/main/java/org/zaproxy/zap/extension/sequence/automation/SequenceActiveScanJob.java:259]: I'll take a look.

**Comment by kingthorin** [comment]
Re-jigged. @thc202, sorry again 😀

File: addOns/sequence/src/main/java/org/zaproxy/zap/extension/sequence/StdActiveScanRunner.java [hunk 2]
```diff
@@ -153,20 +156,32 @@ public ZestResponse runStatement(
...
+                steps.add(stepData);
+                countStepStats(stepData);
             }
         }
 
         return resp;
     }
 
+    private static void countStepStats(SequenceStepData step) {
+        String ascanStep = ExtensionSequenceAutomation.STATS_PREFIX + "ascan.step" + step.getStep();
```
Comment (thc202) [inline addOns/sequence/src/main/java/org/zaproxy/zap/extension/sequence/StdActiveScanRunner.java:176]: Looks good but we should now use a more generic constant for these outside the automation.

**Comment by thc202** [comment]
I think it could still keep the error and execution count for the active scan job.

**Comment by kingthorin** [comment]
Done & done

File: addOns/sequence/src/main/java/org/zaproxy/zap/extension/sequence/StdActiveScanRunner.java [hunk 3]
```diff
@@ -107,6 +110,7 @@ public String run(ZestScript script, Map<String, String> params)
                     ZestInvalidCommonTestException,
                     ZestAssignFailException,
                     ZestClientFailException {
+        Stats.incCounter(STATS_PREFIX + "ascan.scan");
```
[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/sequence/src/main/java/org/zaproxy/zap/extension/sequence/StdActiveScanRunner.java:113]: Yup missed a period

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/sequence/src/main/java/org/zaproxy/zap/extension/sequence/StdActiveScanRunner.java:113]: Drop the ascan too?

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/sequence/src/main/java/org/zaproxy/zap/extension/sequence/StdActiveScanRunner.java]: Done

File: addOns/sequence/CHANGELOG.md [hunk 4]
```diff
@@ -10,6 +10,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
   - `sequence-activeScan` to active scan sequences.
 - Data for reporting.
 - Sequence active scan policy.
+- Stats for import and active scan automation.
```
Comment (thc202) [inline addOns/sequence/CHANGELOG.md:13]: This needs tweak as it's no longer just for automation.

File: addOns/callhome/CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## Unreleased
 ### Added
 - Network stats to telemetry.
+- Sequence automations stats to telemetry.
```
Comment (thc202) [inline addOns/callhome/CHANGELOG.md:9]: I guess same here.

**Review by psiinon — APPROVED** [review]

**Pull request merged**